wal.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package wal;
+option go_package = "internal/wal";
+
+// Version information for the WAL format
+enum WALVersion {
```

Comment:
We can skip this altogether. We may not need to store the WAL version given we can make sure protobuf are always kept backward compatible. what do you tthink?